### DISCUSSION TITLE
Using $loop Variable in Table Component

### DISCUSSION THREAD

**Discussion by elnasnato** [tasteui/tasteui#716] [Q&A]
Can we use $loop variable when using table component to create automatic numbering? I usually use a method like this in view:
@foreach(...)
{{ $loop->index + $rows->firstItem() }}
@endforeach
can you show me the hint please? Thanks!

**Comment by devajmeireles** [comment]
Hey, @elnasnato ! Sorry for the delay in responding. Actually, you are completely free to use `$loop` inside the table, but even without looking at the source code, I assume the correct one is `$this->loop`.

**Reply by elnasnato** [reply]
Thank you for taking your time, but sorry this is my actual code :
error that appears :

htmlspecialchars(): Argument # 1 ($string) must be of type string, stdClass given

![code](https://github.com/user-attachments/assets/d42e1936-2164-4f3b-957c-a7f84a4c8b8f)

numbering should goes here
![code2](https://github.com/user-attachments/assets/f4bdc12b-bac6-4808-86b1-8231375f7337)
